### ISSUE TITLE
How to permanently delete all 0install folders in AppData FFS?!

### ISSUE THREAD

**Issue by FuckZeoInstall** [0install/0install-win#155]
Hi, 

I thought ZeroInstall was all about NOT installing anything other than executables, à la PortableApps, but I was sorely mistaken. Now I find myself with an undeletable folder in my AppData\Local folder that nothing will rid me of. 

I've tried command line, LockHunter, takeown, etc... No dice. 

Can someone tell me how to remove this folder  for good WITHOUT INSTALLING ZEROINSTALL  because I never want to have anything to do with sh*t ever again? 

This is bordering on malware install, you ask me, because i should decide what remains on my hard drive and what doesn't.

**Comment by bastianeicher** [comment]
Directories inside the Zero Install cache need to be immutable to make sure their contents always matches the hash which is their name. To prevent apps from changing things in these directories after they are first extracted from archives (e.g., self updates) the directories are made read-only using NTFS ACLs.

0install automatically puts a file named `How to delete.txt` in the directory explaining this. Since *this* file is not read-only you may already have deleted it though.

To remove the directory without using 0install you can run the following in PowerShell:

    icacls $env:localappdata\0install.net /t /q /c /reset
    rm -Recurse $env:localappdata\0install.net

I’m sorry to hear you were disappointed by 0install. However, I’d encourage you to rethink your tone. After all, this is a free volunteer-developed tool and you came here asking for help with something which you, frankly, could have figured yourself by reading the docs.

Nonetheless, I’d be happy to hear any suggestions you might have how to make 0install appear less malware-y to you. I honestly believe this comes down to a misunderstanding which could probably be avoided with better messages or docs.

**Comment by FuckZeoInstall** [comment]
Hi, and thanks for the quick reply! 

What you suggested worked flawlessly, thank you! As for the rest, I understand that the application needs to keep its own settings somewhere "safe", hence its writing them in Appdata\Local and \Roaming (for the latter, deleting the folder in question didn't pose any issues), but for me, and unless I completely misunderstood the point of ZeroInstall, it seems a little bit over-the-top to prevent a local admin from removing those settings after deleting the ZeroInstall application he or she took for a test drive on their PC. 

I'm sorry if my remark about this behavior being tantamount to malware offended you, because it wasn't my intention at all. It's just that I didn't expect that kind of behavior from a program that professes not to install anyting on my PC. 

I've used PortableApps programs a lot over the years, and aside from some entries being written to Registry, they didn't leave any traces behind when I deleted them. 

I expected the same behavior from ZeroInstall, but maybe I got it all wrong. In any case, thanks a lot for your help! 

Stay safe!
